plugins/PrivacyManager/Config.php

```diff
@@ -67,43 +81,93 @@ public function __get($name)
         return $this->getFromTrackerCache($name, $this->properties[$name]);
     }
 
-    public static function prefix(string $optionName): string
+    public function prefix(string $optionName, bool $addIdSite = true): string
     {
-        return 'PrivacyManager.' . $optionName;
+        // if requested, adding the site ID in the middle to have all the site-specific settings together
+        return 'PrivacyManager.' . (($addIdSite && $this->idSite) ? "idSite($this->idSite)." : '') . $optionName;
```

Comment:
We should delete those new site specific options when a site is deleted.

The event (see e.g. [Goals plugin](https://github.com/matomo-org/matomo/blob/be81d28b79efc6795c07aa14535e30a9b6b1df7a/plugins/Goals/Goals.php#L105)) os readily available and we can prevent our options able from bloating more than necessary.

Reply:
Good suggestion, implemented.